libraries/AP_NavEKF2/AP_NavEKF2_PosVelFusion.cpp

```diff
@@ -942,7 +942,7 @@ void NavEKF2_core::selectHeightForFusion()
     if (extNavUsedForPos) {
         // always use external vision as the height source if using for position.
         activeHgtSource = HGT_SOURCE_EXTNAV;
-    } else if (_rng && ((frontend->_useRngSwHgt > 0) && (frontend->_altSource == 1)) && (imuSampleTime_ms - rngValidMeaTime_ms < 500)) {
+    } else if (_rng && (frontend->_useRngSwHgt > 0) && (imuSampleTime_ms - rngValidMeaTime_ms < 500)) {
```

Comment:
This should be split into two cases. One when frontend->_altSource == 1 which always ruses the range finder as the active height source and one when frontend->_useRngSwHgt > 0  and frontend->_altSource != which switches between the primary height source and range finder using the height and speed criteria.

Reply:
> ... and one when frontend->_useRngSwHgt > 0 and frontend->_altSource != which ...

I Think this should be:
... and one when frontend->_useRngSwHgt > 0 and frontend->_altSource != 1 which ...